delta/plugins/graph-analytics/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/graph/analytics/indexing/GraphAnalyticsSinkSuite.scala

```diff
@@ -0,0 +1,174 @@
+  val resource3         = iri"http://localhost/resource3"
+  val file1             = iri"http://localhost/file1"
+
+  private def loadExpanded(path: String): UIO[ExpandedJsonLd] =
+    ioJsonContentOf(path)
+      .flatMap { json =>
+        Task.fromEither(ExpandedJsonLd.expanded(json))
+      }
+      .memoizeOnSuccess
+      .hideErrors
+
+  private def getTypes(expandedJsonLd: ExpandedJsonLd): UIO[Option[Set[Iri]]] =
+    UIO.pure(expandedJsonLd.cursor.getTypes.toOption)
+
+  private val expanded1 = loadExpanded("expanded/resource1.json")
+  private val expanded2 = loadExpanded("expanded/resource2.json")
+
+  val findRelationship: Iri => UIO[Option[Set[Iri]]] = {
+    case `resource1` =>
+      expanded1.flatMap(getTypes)
+    case `resource2` =>
+      expanded2.flatMap(getTypes)
+    case `resource3` =>
+      UIO.some(Set(iri"https://neuroshapes.org/Trace"))
+    case _           => UIO.none
+  }
+
+  test("Create the update script and the index") {
+    for {
+      script  <- scriptContent
+      _       <- client.createScript(updateRelationshipsScriptId, script)
+      mapping <- graphAnalyticsMappings
+      _       <- client.createIndex(index, Some(mapping), None).assert(true)
+    } yield ()
+  }
+
+  test("Push index results") {
+    def toIndex(id: Iri, io: UIO[ExpandedJsonLd]) = {
+      for {
+        expanded <- io
+        types    <- getTypes(expanded)
+        doc      <- JsonLdDocument.fromExpanded(expanded, findRelationship)
+      } yield {
+        val result = GraphAnalyticsResult.Index(
+          id,
+          1,
+          types.getOrElse(Set.empty),
+          Instant.EPOCH,
+          Anonymous,
+          Instant.EPOCH,
+          Anonymous,
+          doc
+        )
+        SuccessElem(Resources.entityType, id, Some(project), Instant.EPOCH, Offset.start, result, 1)
+      }
+    }
+
+    for {
+      r1                <- toIndex(resource1, expanded1)
+      r2                <- toIndex(resource2, expanded2)
+      r3                 = SuccessElem(
+                             Resources.entityType,
+                             resource3,
+                             Some(project),
+                             Instant.EPOCH,
+                             Offset.start,
+                             GraphAnalyticsResult.Noop,
+                             1
+                           )
+      chunk              = Chunk.seq(List(r1, r2, r3))
+      // We expect no error
+      _                 <- sink(chunk).assert(chunk.map(_.void))
+      // Elasticsearch should have taken into account the bulk query and the update query
+      expectedDocuments <- ioJsonContentOf("indexed-documents.json")
+      sort               = SortList(Sort("@id") :: Nil)
+      _                 <- client
+                             .search(JsonObject.empty, Set(index.value), Uri.Query.Empty)(sort)
+                             .map(_.removeKeys("took", "_shards"))
+                             .eventually(expectedDocuments)
```

Comment:
Perhaps we can assert on the things we need.. like the hits containing the expected document ids?

Another possibility is making an alias for _.removeKeys("took", "_shards") to explain that you don't want to compare the ES metadata, as it's not clear to someone unfamiliar with ES that this is why you remove those fields

Reply:
The result is a bit large but there is only two documents in the response and both are important.
About the ES testing, it can be included in our thinking on how we design tests but the way to go could be at some point to rely on `elastic4s` at least for the tests

Reply:
In the long run, we should avoid to compare json payloads as much as we do